.gitlab-ci.yml

```
@@ -290,9 +293,17 @@ variables:
 .on_deploy_a6_manual:
   - <<: *if_not_version_6
     when: never
+  - <<: *if_tagged_deploy
+    when: manual
+    allow_failure: true
+    variables:
+      AGENT_REPOSITORY: agent
   - <<: *if_deploy
     when: manual
     allow_failure: true
+    variables:
+      AGENT_REPOSITORY: agent-dev
+      IMG_REGISTRIES: dev
```

The previous behavior was the following: if the release branch is `nightly`, send to `agent-dev`, otherwise send to `agent`: https://github.com/DataDog/datadog-agent/blob/bea56884ae152e63f1aea5d726a70f69a3234544/.gitlab/deploy_7/docker.yml#L57-L59

I think we want to keep that. To do this, you can do:
```suggestion
.on_deploy_a6_manual:
  - <<: *if_not_version_6
    when: never
  - <<: *if_not_deploy
    when: never
  - <<: *if_not_stable_or_beta_repo_branch
    when: manual
    allow_failure: true
    variables:
      AGENT_REPOSITORY: agent-dev
      IMG_REGISTRIES: dev
  - when: manual
    allow_failure: true
    variables:
      AGENT_REPOSITORY: agent
```

You'll need to define:
```
.if_not_deploy: &if_not_deploy
  if: $DEPLOY_AGENT != "true"
```

instead of `.if_tagged_commit`